graphium/trainer/predictor.py

```diff
@@ -449,35 +447,47 @@ def flag_step(self, batch: Dict[str, Tensor], step_name: str, to_cpu: bool) -> D
 
     def on_train_batch_start(self, batch: Any, batch_idx: int) -> Optional[int]:
         self.train_batch_start_time = time.time()
+        self.is_log_train_metrics = (self.metrics_every_n_train_steps is None) or (
+            (batch_idx + 1) % self.metrics_every_n_train_steps != 0
+        )
```

Comment:
Does this mean if `(batch_idx + 1) % self.metrics_every_n_train_steps != 0`, `self.is_log_train_metrics` is true and we only log the metrics in `concatenated_metrics_logs` which is global loss, `batch_tput` stuff. When `(batch_idx + 1) % self.metrics_every_n_train_steps == 0`, we log `preds`, `targets` as well as individual losses? This sounds good to me.

Reply:
Yes exactly!

Reply:
I changed `batch_idx + 1` --> `batch_idx` because I think it's a good idea to log the first batch.